sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/SessionReceiverManager.cs

```diff
@@ -287,7 +295,7 @@ public override async Task ReceiveAndProcessMessagesAsync(CancellationToken proc
                 }
             }
             catch (Exception ex)
-            when (!(ex is TaskCanceledException) ||
+            when (ex is not TaskCanceledException ||
```

Comment:
I found this flow difficult to read.   It took me a few passes to pick up on 299-301 still being part of the `where`.